overlord/hookstate/ctlcmd/is_connected.go

```diff
@@ -87,6 +91,15 @@ func (c *isConnectedCommand) Execute(args []string) error {
 		return fmt.Errorf("internal error: cannot get connections: %s", err)
 	}
 
+	var otherSnapName string
+	if c.Pid != 0 {
+		otherSnapName, err = cgroupSnapNameFromPid(c.Pid)
+		// FIXME: should we treat non-snap processes as connected?
```

Comment:
I don't think so. We could have a tri-state answer: yes, no, not-a-snap

Reply:
I agree.  I think distinct non-zero error codes for "not connected", "not connected, but pid is classic confined", and "not a snap" would be workable.

That way, callers that treat the exit status as a simple boolean will get the expected behaviour of determining whether a connection exists, while still allowing things like Pulse Audio or CUPS to service requests from the host system or classic snaps.